wms/models/datasets/sgrid.py

```diff
@@ -457,7 +458,141 @@ def _spatial_data_subset(self, data, spatial_index):
         columns = spatial_index[1, :]
         data_subset = data[rows, columns]
         return data_subset
-
+    
+    def _avg_to_cell_center(self, data_array, avg_dim, vectorstep):
+        """
+        Given a two-dimensional numpy.array, average
+        adjacent row values (avg_dim=1) or adjacent
+        column values (avg_dim=0) to the grid cell
+        center.
+        
+        :param data_array: 2-dimensional data
+        :type data_array: numpy.array
+        :param int avg_dim: integer specify array axis to be averaged
+        :return: averages
+        :rtype: numpy.array
+        
+        """
+        if avg_dim == 0:
+            da = np.transpose(data_array)
+        else:
+            da = data_array
+        da_trim_low = da[:, 1:][::vectorstep, ::vectorstep]
```

Comment:
@ayan-usgs as we discussed in https://github.com/sgrid/pysgrid/issues/74#issuecomment-173249585 I belive that, if we implement lazy `.avg_{center,node}` methods, we can just append the `trim_{center,node}` operation to the subset slice and dramatically reduce the amount of data downloaded in `pysgrid` itself.  (No need for this PR here.)